### ISSUE TITLE
SemanticsController.simulatedAccessibilityTraversal broken in multiple views

### ISSUE THREAD

**Issue by Piinks** [flutter/flutter#143405]
Found in
- Issue https://github.com/flutter/flutter/issues/143173
- PR https://github.com/flutter/flutter/pull/143386

In SemanticsController.simulatedAccessibilityTraversal, the `start` and `end` parameters are deprecated. They are replaced by `startNode` and `endNode`, but writing a test for this in the context of multiple views fails. 

I wrote such a test in `packages/flutter_test/test/multi_view_controller_test.dart`

```dart
testWidgets('simulatedAccessibilityTraversal - startNode and endNode in same view', (WidgetTester tester) async {
    tester.binding.ensureSemantics();
    await pumpViews(tester: tester);
    expect(
      tester.semantics.simulatedAccessibilityTraversal(
        startNode: find.semantics.byValue('View2Child1'),
        endNode: find.semantics.byValue('View2Child3'),
      ).map((SemanticsNode node) => node.label),
      <String>[
        'View2Child1',
        'View2Child2',
        'View2Child3',
      ],
    );
  });
```

but it failed with 

```The following StateError was thrown running a test:
Bad state: No element

When the exception was thrown, this was the stack:
#0      Iterable.single (dart:core/iterable.dart:674:25)
#1      SemanticsController.simulatedAccessibilityTraversal (package:flutter_test/src/controller.dart:207:57)
```

This is coming from here when we are trying to look up the SemanticsOwner:

https://github.com/flutter/flutter/blob/efd1d1f536ece87fc8c3437879f7f95f12e0d184/packages/flutter_test/lib/src/controller.dart#L202

The new API works in the single view world, so I am going to move forward with that PR to resolve other issues unrelated to this one. 

cc @goderbauer
